### ISSUE TITLE
Error calling find_maxima()

### ISSUE THREAD

**Issue by danielaik** [dwaithe/MaximaFinder#9]
Hello @dwaithe,

I came across error calling `find_maxima()`  as I run `find_maxima.ipynb`

Error messages:

```
../site-packages/findmaxima2d/findmaxima2d.py:53, in find_maxima(img_data, local_max, ntol)
../site-packages/findmaxima2d/cfindmaxima2d.pyx:227, in findmaxima2d.cfindmaxima2d.find_maxima()

 OverflowError: Python integer 4294967293 out of bounds for uint8. 
```

I tried both **pip install findmaxima2d==0.0.27** and manual installation directly form the repository version **0.025**. Both gave the same issue.

Any tips?

Thanks in advance,
Daniel

**Comment by danielaik** [comment]
Solved by using older version of `numpy==1.24.4` in version **0.0.27**

When building with version **0.025** (this repo), `np.bool` was deprecated so `return local_max.astype(np.uint8)` or `return local_max.astype(bool)` in place of `return local_max.astype(np.bool)`

```
def find_local_maxima(img_data, size=3):
    """
    Function which finds local maxima in neighbourhood for a grayscale image.

    inputs:
    --------------
    img_data  - 2-D image data.
    size (optional) - Size of neighbourhood to scan, default = 3 (recommended).

    outputs:
    --------------
    local_max - 2-D bool image with local maxima.
    """

    assert img_data.shape != 2, "Your input image is the wrong dimension, it should 2-D."

    # Filter data with maximum filter to find maximum filter response in each neighbourhood
    max_out = ndimage.filters.maximum_filter(img_data, size=3)
    # Find local maxima.
    local_max = np.zeros((img_data.shape))
    local_max[max_out == img_data] = 1
    local_max[img_data == np.min(img_data)] = 0
    # return local_max.astype(np.bool)
    return local_max.astype(np.uint8)
```

**Comment by dwaithe** [comment]
Hi Daniel,
Thanks for your efforts here. I have updated this repository and the pip wheel to use bool, rather than np.bool. Thanks for your patience: pip install findmaxima2d==0.0.28
